### DISCUSSION TITLE
I need to send custom info. Are there any references I can use?

### DISCUSSION THREAD

**Discussion by zerocooldog** [open-telemetry/opentelemetry-java#6763] [Q&A]
I need to send custom info.

I want to send my own custom data to the grpc server, in addition to the defined exporters like traces, metrics, logs.

```
syntax = "proto3";

package opentelemetry.proto.collector.hamonica2.v1;
import "opentelemetry/proto/common/v1/common.proto";

service CommandService {
  rpc Export (CommandRequest) returns (CommandResponse);
}

message CommandRequest{

  string serverType = 1;
  string serviceInstanceId = 2;
  string serviceName = 3;
  string serviceNamespace = 4;
  string command = 5;
  map<string, string>	parameter = 6; 

}

message CommandResponse {
  bytes data = 1; 
  common.v1.ResponseStatus status = 2;
  string message = 3;
}

```

Can I develop based on the reference provided in the following URL?

https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/logs/OtlpGrpcLogRecordExporterBuilder.java

**Comment by jkwatson** [comment]
Are you asking about how to implement a generic GRPC client? If so, I recommend reading the GRPC docs, rather than basing anything on the way we've implemented things. We've chosen to do a lot of very custom code, to optimize things more than would normally be necessary.

**Reply by zerocooldog** [reply]
That's correct. I didn't want to implement it myself because I thought there would be a client transmission method in OpenTelemetry for sending to a gRPC server.

I asked this question in the opentelemetry-java-instrumentation project, but I was directed to ask here. However, the answer seems to be the same. I understand it to mean that I need to implement it separately.

Thank you for your reply.
